service/frontend/workflowHandler.go

```diff
@@ -2176,27 +1827,50 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		token.IsWorkflowRunning = isWorkflowRunning
 		token.PersistenceToken = nil
 	}
+	rawHistoryQueryEnabled := wh.config.SendRawWorkflowHistoryByDomain(getRequest.GetDomain())
+
+	var history *gen.History
+	var historyBlob []*gen.DataBlob
 
-	history := &gen.History{}
-	history.Events = []*gen.HistoryEvent{}
 	if isCloseEventOnly {
 		if !isWorkflowRunning {
-			history, _, err = wh.getHistory(
-				scope,
-				domainID,
-				*execution,
-				lastFirstEventID,
-				nextEventID,
-				getRequest.GetMaximumPageSize(),
-				nil,
-				token.TransientDecision,
-				token.BranchToken,
-			)
-			if err != nil {
-				return nil, wh.error(err, scope)
+			if rawHistoryQueryEnabled {
+				historyBlob, _, err = wh.getRawHistory(
+					scope,
+					domainID,
+					*execution,
+					lastFirstEventID,
+					nextEventID,
+					getRequest.GetMaximumPageSize(),
+					nil,
+					token.TransientDecision,
+					token.BranchToken,
+				)
+				if err != nil {
+					return nil, wh.error(err, scope)
+				}
+
+				// since getHistory func will not return empty history, so the below is safe
+				historyBlob = historyBlob[len(historyBlob)-1 : len(historyBlob)]
+			} else {
+				history, _, err = wh.getHistory(
+					scope,
+					domainID,
+					*execution,
+					lastFirstEventID,
+					nextEventID,
+					getRequest.GetMaximumPageSize(),
+					nil,
+					token.TransientDecision,
+					token.BranchToken,
+				)
+
+				if err != nil {
+					return nil, wh.error(err, scope)
+				}
+				// since getHistory func will not return empty history, so the below is safe
+				history.Events = history.Events[len(history.Events)-1 : len(history.Events)]
 			}
-			// since getHistory func will not return empty history, so the below is safe
-			history.Events = history.Events[len(history.Events)-1 : len(history.Events)]
 			token = nil
 		} else if isLongPoll {
```

Comment:
previously, the else if and else code paths had history and history.Events non-nil - with your code change, they will be nil. Can you fix this to not change existing behavior ?